lib/Runtime/Base/PropertyRecord.h

```diff
@@ -121,8 +121,43 @@ namespace Js
 
         bool Equals(JsUtil::CharacterBuffer<WCHAR> const & str) const
         {
-            return (LEN - 1 == str.GetLength() &&
-                JsUtil::CharacterBuffer<WCHAR>::StaticEquals(buffer, str.GetBuffer(), LEN - 1));
+#ifndef _NTBUILD
+            AssertMsg(false, "Do you really have to use this interface?");
+#endif
+            return Equals(str.GetBuffer(), str.GetLength());
+        }
+
+        bool Equals(JavascriptString * str) const
+        {
+            PropertyString * propString = PropertyString::TryFromVar(str);
+            if (propString == nullptr)
+            {
+                LiteralStringWithPropertyStringPtr * lstr = LiteralStringWithPropertyStringPtr::TryFromVar(str);
+                // propString = lstr->GetOrAddPropertyString();
+                // To enable the line above, waiting for another PR to merge
+                // if you have time, please review #3875
+            }
+
+            if (propString == nullptr)
+            {
+                (LEN - 1 == str->GetLength() &&
+                    JsUtil::CharacterBuffer<WCHAR>::StaticEquals(buffer, str->GetString(), LEN - 1));
```

Comment:
Is this supposed to be assigned to something?

Reply:
Looks like this should be `return Equals(str.GetString(), str.GetLength);` ?